provider/core/src/resource.rs

```diff
@@ -14,6 +14,21 @@ use core::fmt::Write;
 use icu_locid::LanguageIdentifier;
 use writeable::{LengthHint, Writeable};
 
+/// A compact hash of a [`ResourceKey`]. Useful for keys in maps.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone, Hash)]
+#[repr(transparent)]
+pub struct ResourceKeyHash([u8; 4]);
```

Comment:
Suggestion: A much easier thing to do to make this ULE-compatible is simply

```rust
impl AsULE for ResourceKeyHash {
    type ULE = RawBytesULE<4>;
    // ...
}
```

And then you don't even need the getter function.

Reply:
Or you can do

```rust
impl ULE for ResourceKeyHash { ... }

impl AsULE for ResourceKeyHash {
    type ULE = Self;
}
```

Reply:
CC @Manishearth